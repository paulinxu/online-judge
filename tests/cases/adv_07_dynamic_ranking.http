###
# Send request:
POST http://127.0.0.1:12345/users HTTP/1.1
content-type: application/json

{"name":"user1"}
###
# Got response:
# HTTP 200
# content-length: 23
# content-type: application/json
# date: Thu, 25 Jul 2024 15:32:39 GMT
# 
# {"id":1,"name":"user1"}
###
# Send request:
POST http://127.0.0.1:12345/users HTTP/1.1
content-type: application/json

{"name":"user2"}
###
# Got response:
# HTTP 200
# content-length: 23
# content-type: application/json
# date: Thu, 25 Jul 2024 15:32:39 GMT
# 
# {"id":2,"name":"user2"}
###
# Send request:
POST http://127.0.0.1:12345/jobs HTTP/1.1
content-type: application/json

{"contest_id":0,"language":"Rust","problem_id":0,"source_code":"fn main() { std::thread::sleep(std::time::Duration::from_millis(500)); println!(\"Hello World!\"); }","user_id":0}
###
# Got response:
# HTTP 200
# content-length: 470
# content-type: application/json
# date: Thu, 25 Jul 2024 15:32:39 GMT
# 
# {"cases":[{"id":0,"info":"","result":"Compilation Success","time":0},{"id":1,"info":"","result":"Accepted","time":500707}],"created_time":"2024-07-25T15:32:40.055406987Z","id":0,"result":"Accepted","score":50.0,"state":"Finished","submission":{"contest_id":0,"language":"Rust","problem_id":0,"source_code":"fn main() { std::thread::sleep(std::time::Duration::from_millis(500)); println!(\"Hello World!\"); }","user_id":0},"updated_time":"2024-07-25T15:32:40.745433167Z"}
###
# Send request:
POST http://127.0.0.1:12345/jobs HTTP/1.1
content-type: application/json

{"contest_id":0,"language":"Rust","problem_id":0,"source_code":"fn main() { std::thread::sleep(std::time::Duration::from_millis(1000)); println!(\"Hello World!\"); }","user_id":1}
###
# Got response:
# HTTP 200
# content-length: 472
# content-type: application/json
# date: Thu, 25 Jul 2024 15:32:40 GMT
# 
# {"cases":[{"id":0,"info":"","result":"Compilation Success","time":0},{"id":1,"info":"","result":"Accepted","time":1001116}],"created_time":"2024-07-25T15:32:40.746717529Z","id":1,"result":"Accepted","score":50.0,"state":"Finished","submission":{"contest_id":0,"language":"Rust","problem_id":0,"source_code":"fn main() { std::thread::sleep(std::time::Duration::from_millis(1000)); println!(\"Hello World!\"); }","user_id":1},"updated_time":"2024-07-25T15:32:41.930967242Z"}
###
# Send request:
POST http://127.0.0.1:12345/jobs HTTP/1.1
content-type: application/json

{"contest_id":0,"language":"Rust","problem_id":0,"source_code":"fn main() { std::thread::sleep(std::time::Duration::from_millis(100)); println!(\"Hello World!\"); }","user_id":2}
###
# Got response:
# HTTP 200
# content-length: 470
# content-type: application/json
# date: Thu, 25 Jul 2024 15:32:41 GMT
# 
# {"cases":[{"id":0,"info":"","result":"Compilation Success","time":0},{"id":1,"info":"","result":"Accepted","time":100883}],"created_time":"2024-07-25T15:32:41.932558936Z","id":2,"result":"Accepted","score":50.0,"state":"Finished","submission":{"contest_id":0,"language":"Rust","problem_id":0,"source_code":"fn main() { std::thread::sleep(std::time::Duration::from_millis(100)); println!(\"Hello World!\"); }","user_id":2},"updated_time":"2024-07-25T15:32:42.220345770Z"}
###
# Send request:
GET http://127.0.0.1:12345/contests/0/ranklist HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 1890
# content-type: application/json
# date: Thu, 25 Jul 2024 15:32:42 GMT
# 
# [{"competitive_score_sum":0.0,"highest_scores":[50.0],"latest_scores":[50.0],"latest_submission":"2024-07-25T15:32:41.932558936Z","rank":1,"score":100,"scores":[100.0],"shortest_times":[[100883,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807]],"submission_count":1,"user":{"id":2,"name":"user2"}},{"competitive_score_sum":0.0,"highest_scores":[50.0],"latest_scores":[50.0],"latest_submission":"2024-07-25T15:32:40.055406987Z","rank":2,"score":60,"scores":[60.074055],"shortest_times":[[500707,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807]],"submission_count":1,"user":{"id":0,"name":"root"}},{"competitive_score_sum":0.0,"highest_scores":[50.0],"latest_scores":[50.0],"latest_submission":"2024-07-25T15:32:40.746717529Z","rank":3,"score":55,"scores":[55.03853],"shortest_times":[[1001116,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807]],"submission_count":1,"user":{"id":1,"name":"user1"}}]
