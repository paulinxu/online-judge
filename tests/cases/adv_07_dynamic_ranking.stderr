[2024-07-25T15:32:39Z INFO  actix_server::builder] starting 16 workers
[2024-07-25T15:32:39Z INFO  actix_server::server] Actix runtime found; starting in Actix runtime
[2024-07-25T15:32:40Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET / HTTP/1.1" 404 0 "-" "-" 0.000160
[2024-07-25T15:32:40Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /users HTTP/1.1" 200 23 "-" "-" 0.000287
[2024-07-25T15:32:40Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /users HTTP/1.1" 200 23 "-" "-" 0.000094
[2024-07-25T15:32:40Z INFO  oj::jobs_module::function_post_jobs] Time used by 0: 500707
[2024-07-25T15:32:40Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /jobs HTTP/1.1" 200 470 "-" "-" 0.690236
[2024-07-25T15:32:41Z INFO  oj::jobs_module::function_post_jobs] Time used by 1: 1001116
[2024-07-25T15:32:41Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /jobs HTTP/1.1" 200 472 "-" "-" 1.184511
[2024-07-25T15:32:42Z INFO  oj::jobs_module::function_post_jobs] Time used by 2: 100883
[2024-07-25T15:32:42Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /jobs HTTP/1.1" 200 470 "-" "-" 0.288057
[2024-07-25T15:32:42Z INFO  oj::contest_module::function_get_contests] 60
[2024-07-25T15:32:42Z INFO  oj::contest_module::function_get_contests] 55
[2024-07-25T15:32:42Z INFO  oj::contest_module::function_get_contests] 100
[2024-07-25T15:32:42Z INFO  oj::contest_module::function_get_contests] [RankInfo { user: User { id: 2, name: "user2" }, rank: 1, scores: [100.0], highest_scores: [50.0], latest_scores: [50.0], competitive_score_sum: 0.0, shortest_times: [[100883, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807]], latest_submission: 2024-07-25T15:32:41.932558936Z, score: 100, submission_count: 1 }, RankInfo { user: User { id: 0, name: "root" }, rank: 2, scores: [60.074055], highest_scores: [50.0], latest_scores: [50.0], competitive_score_sum: 0.0, shortest_times: [[500707, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807]], latest_submission: 2024-07-25T15:32:40.055406987Z, score: 60, submission_count: 1 }, RankInfo { user: User { id: 1, name: "user1" }, rank: 3, scores: [55.03853], highest_scores: [50.0], latest_scores: [50.0], competitive_score_sum: 0.0, shortest_times: [[1001116, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807]], latest_submission: 2024-07-25T15:32:40.746717529Z, score: 55, submission_count: 1 }]
[2024-07-25T15:32:42Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /contests/0/ranklist HTTP/1.1" 200 1890 "-" "-" 0.000305
